### PR TITLE
Automatically Install & Activate the Import Users from CSV plugin

### PR DESCRIPTION
**Pull request by JarrydLong**

### CODE REVIEW

File: pmpro-import-users-from-csv.php [hunk 1]
```diff
@@ -68,11 +68,110 @@ function pmproiufcsv_admin_notice(){
 	
 	?>
     <div class="notice notice-warning">
-        <p><?php printf( __( 'In order for <strong>Paid Memberships Pro - Import Users from CSV</strong> to function correctly, you must also install the <a href="%s">Import Users from CSV</a> plugin.', 'pmproiufcsv' ), esc_url( admin_url( 'plugin-install.php?tab=search&s=import+users+from+csv+andrew+lima' ) ) ); ?></p>
+        <p><?php printf( __( 'In order for <strong>Paid Memberships Pro - Import Users from CSV</strong> to function correctly, you must also install the <a href="%s">Import Users from CSV</a> plugin.', 'pmproiufcsv' ), esc_url( admin_url( 'tools.php?page=pmproiufcsv-install-plugin' ) ) ); ?></p>
```
Comment (sc0ttkclark) [inline pmpro-import-users-from-csv.php:71]: We could bypass the need for a hidden page for installing and just use a link like this:

```php
$url = wp_nonce_url(
	add_query_arg(
		[
			'action' => 'install-plugin',
			'plugin' => 'import-users-from-csv',
			'from'   => 'import',
		],
		admin_url( 'update.php' )
	),
	'install-plugin_import-users-from-csv'
);
```

and then pass that into line 71 for the link.

File: pmpro-import-users-from-csv.php [hunk 2]
```diff
@@ -68,11 +68,110 @@ function pmproiufcsv_admin_notice(){
...
+	include_once ABSPATH . 'wp-admin/includes/class-wp-upgrader.php';
+	wp_cache_flush();
+
+	$upgrader = new Plugin_Upgrader();
+	$upgraded = $upgrader->upgrade( $plugin_slug );
+
+	return $upgraded;
+}
+
+
```
Comment (sc0ttkclark) [inline pmpro-import-users-from-csv.php:174]: Let's go the simpler route and use the built-in plugin installation page

File: pmpro-import-users-from-csv.php (hunk 1 above)
Comment (sc0ttkclark) [reply pmpro-import-users-from-csv.php:71]: Before showing the link, we also need to check if they have the correct capability to install plugins.

`current_user_can( 'install_plugins' )`

If the plugin is already installed but needs to be activated, we can use this link:

```php
$url = wp_nonce_url(
	add_query_arg(
		[
			'action' => 'activate',
			'plugin' => 'import-users-from-csv',
		],
		admin_url( 'plugins.php' )
	),
	'activate-plugin_import-users-from-csv'
);
```

Before showing the activate link itself, we should also check for access rights by checking `current_user_can( 'activate_plugin', 'import-users-from-csv' )`

**Review by sc0ttkclark — CHANGES_REQUESTED** [review]
Needs a new approach

File: pmpro-import-users-from-csv.php (hunk 2 above)
Comment (JarrydLong) [reply pmpro-import-users-from-csv.php:174]: I'm not sure I understand, which plugin installation page?

**Review by sc0ttkclark — APPROVED** [review]

File: pmpro-import-users-from-csv.php [hunk 3]
```diff
@@ -75,16 +75,65 @@ function pmproiufcsv_admin_notice(){
...
+	if ( pmproiufcsv_is_plugin_installed( 'import-users-from-csv/import-users-from-csv.php' ) ) {
+		$action_url = wp_nonce_url(
+			add_query_arg(
+				[
+					'action' => 'activate',
+					'plugin' => 'import-users-from-csv',
+				],
+				admin_url( 'plugins.php' )
+			),
+			'activate-plugin_import-users-from-csv'
```
Comment (sc0ttkclark) [inline pmpro-import-users-from-csv.php:100]:
```suggestion
					'plugin' => 'import-users-from-csv/import-users-from-csv.php',
				],
				admin_url( 'plugins.php' )
			),
			'activate-plugin_import-users-from-csv/import-users-from-csv.php'

```

**Pull request merged**